src/daft-core/src/array/ops/approx_count_distinct.rs

```diff
@@ -0,0 +1,69 @@
+use std::collections::HashSet;
+
+use arrow2::array::PrimitiveArray;
+use common_error::DaftResult;
+
+use crate::array::ops::as_arrow::AsArrow;
+use crate::array::ops::DaftApproxCountDistinctAggable;
+use crate::datatypes::UInt64Array;
+use crate::utils::identity_hash_set::IdentityBuildHasher;
+
+impl DaftApproxCountDistinctAggable for UInt64Array {
+    type Output = DaftResult<UInt64Array>;
+
+    fn approx_count_distinct(&self) -> Self::Output {
+        let mut set = HashSet::<_, IdentityBuildHasher>::default();
+        match self.validity() {
+            Some(validity) => {
+                for (index, &value) in self.as_arrow().values_iter().enumerate() {
```

Comment:
you should be able to do `self.as_iter()` which will return Option<u64> so you dont have to manually call `get_bit(index)`

Reply:
Updated!